File: crowdfunding-frontend/src/services/fileService.js
import api from './api';

export const fileService = {
  uploadDocument: (file) => {
    const formData = new FormData();
    formData.append('file', file);
    return api.post('/files/upload/document', formData);
  },
  uploadImage: (file) => {
    const formData = new FormData();
    formData.append('file', file);
    return api.post('/files/upload/image', formData);
  },
};

File: crowdfunding-backend/src/main/java/com/example/project/service/interfaces/FileService.java
package com.example.project.service.interfaces;

import org.springframework.web.multipart.MultipartFile;

public interface FileService {
    
    /**
     * Upload an image file, resize/optimize it, and save it locally.
     * @param file the multipart file from client
     * @return the generated URL path to access the file
     */
    String uploadAndOptimizeImage(MultipartFile file);

    /**
     * Upload a KYC document (PDF or image) and save it as-is.
     * @param file the multipart file from client
     * @return the generated URL path to access the file
     */
    String uploadDocument(MultipartFile file);
}

